### PR TITLE
LOOP-2974 Support pod time zone change

### PR DESCRIPTION
**Pull request by ps2**
https://tidepool.atlassian.net/browse/LOOP-2974

### CODE REVIEW

File: DashKit/DashPumpManager.swift [hunk 1]
```diff
@@ -556,7 +554,8 @@ open class DashPumpManager: PumpManager {
             return
         }
         
-        suspendDelivery { (error) in
+        let reminder = try! StopProgramReminder(value: StopProgramReminder.maxSuspendDuration)
```
Comment (darinkrauss) [inline DashKit/DashPumpManager.swift:557]: Will the `!` ever fail?

File: DashKit/DashPumpManagerState.swift [hunk 2]
```diff
@@ -289,7 +304,8 @@ extension DashPumpManagerState: CustomDebugStringConvertible {
             "* lowReservoirReminderValue: \(lowReservoirReminderValue)",
             "* scheduledExpirationReminderOffset: \(String(describing: scheduledExpirationReminderOffset))",
             "* podAttachmentConfirmed: \(podAttachmentConfirmed)",
-            "* confidenceRemindersEnabled: \(confidenceRemindersEnabled)"
```
Comment (rickpasetto) [inline DashKit/DashPumpManagerState.swift:292]: [nit] did you also want to add `activeAlerts`?

File: DashKit/DashPumpManager.swift [hunk 3]
```diff
@@ -1571,55 +1588,68 @@ extension DashPumpManager: PodCommManagerDelegate {
...
+        }
+        
+        self.mutateState { (state) in
+            state.activeAlerts.insert(alert)
+        }
+    }
+    
+    func retractAlert(alert: PumpManagerAlert) {
+        pumpDelegate.notify { (delegate) in
+            let identifier = Alert.Identifier(managerIdentifier: self.managerIdentifier, alertIdentifier: alert.alertIdentifier)
```
Comment (darinkrauss) [inline DashKit/DashPumpManager.swift]: [nit] Generating identifier inside `notify` is inconsistent with the rest of this new code.

File: DashKit/DashPumpManagerState.swift [hunk 4]
```diff
@@ -187,12 +190,23 @@ public struct DashPumpManagerState: RawRepresentable, Equatable {
             self.finishedDoses = []
         }
         
-        if let rawActiveAlerts = rawValue["activeAlerts"] as? PodAlerts.RawValue {
-            self.activeAlerts = PodAlerts(rawValue: rawActiveAlerts)
+        if let rawActiveAlerts = rawValue["activeAlerts"] as? [PumpManagerAlert.RawValue] {
+            self.activeAlerts = []
```
Comment (darinkrauss) [inline DashKit/DashPumpManagerState.swift]: [nit] Consider moving `self.activeAlerts = []` before `if` statement and getting rid of `else` clause.

Or consider changing everything in the `if` block to just `self.activeAlerts = rawActiveAlerts.compactMap { PumpManagerAlert(rawValue: $0) }`

File: DashKit/DashPumpManagerState.swift [hunk 5]
```diff
@@ -187,12 +190,23 @@ public struct DashPumpManagerState: RawRepresentable, Equatable {
...
+            for rawAlert in rawActiveAlerts {
+                if let alert = PumpManagerAlert(rawValue: rawAlert) {
+                    self.activeAlerts.insert(alert)
+                }
+            }
         } else {
             self.activeAlerts = []
         }
         
+        if let acknowledgedTimeOffsetAlert = rawValue["acknowledgedTimeOffsetAlert"] as? Bool {
```
Comment (darinkrauss) [inline DashKit/DashPumpManagerState.swift]: [nit] Consider changing to `self.acknowledgedTimeOffsetAlert = rawValue["acknowledgedTimeOffsetAlert"] as? Bool ?? false`

**Review by darinkrauss — APPROVED** [review]
A few nits, but otherwise LGTM!

File: DashKitUI/Views/DashSettingsView.swift [hunk 6]
```diff
@@ -19,11 +19,18 @@ struct DashSettingsView: View  {
...
+   
+   @State private var showSyncTimeOptions = false;
 
    @Environment(\.guidanceColors) var guidanceColors
    @Environment(\.insulinTintColor) var insulinTintColor
-
+   
    weak var navigator: DashUINavigator?
+   
+   var pumpTimeIsOffset: Bool {
```
Comment (rickpasetto) [inline DashKitUI/Views/DashSettingsView.swift]: [nit] Hm...would've expected this policy check to be in the `viewModel` (and under unit test...)

**Review by rickpasetto — APPROVED** [review]
One suggestion for moving code to the ViewModel, and a general request for unit tests covering the new code.  Other than that, looks good to me.

File: DashKit/DashPumpManager.swift (hunk 1 above)
Comment (ps2) [reply DashKit/DashPumpManager.swift:557]: No, not unless the SDK was horribly broken.

**Pull request merged**